src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -737,9 +739,11 @@ public static DiagnosticPosition diagnosticPositionFor(final Symbol sym, final L
 
     private static class DeclScanner extends TreeScanner {
         final Symbol sym;
+        final Predicate<? super JCTree> filter;
 
-        DeclScanner(final Symbol sym) {
+        DeclScanner(final Symbol sym, Predicate<? super JCTree> filter) {
```

Comment:
same here, better to keep the original declaration and add an overloaded one with the new argument, the old method can invoke the new one